_posts/2020-09-09-codebuild-reports-cypress-e2e-tests.md

```diff
@@ -0,0 +1,146 @@
+---
+layout: post
+title: "Adding CodeBuild reports to Cypress e2e tests using CDK"
+date: 2020-09-09
+tags: aws codebuild reports testing devops
+author: Natalie Laing
+image: /img/reports.png
+---
+
+![Generic reports](/img/reports.png)
+
+## Introduction
+
+I was recently working on a project where the project owner wanted to be able to visualize our tests in one place without having to navigate through the codebase.
+
+This enabled us to visualize the following in our report group:
+
+- Pass rate
+- Average report duration
+- Average number of test cases run
+- Number of reports generated
+
+CodeBuild reports allow you to drill deeper into each report generated by visualizing:
+
+- Pass rate
+- Report duration
+- When the report was created
+- Individual tests which are filterable by status
+
+CodeBuild stores your test results for 30 days. Test results can be exported to S3 if you need them for more than 30 days.
+
+![CodeBuild report directory](/img/CodeBuild-Reports-Finder.png)
+
+## Set up your infrastructure:
+
+Firstly you will need to set up your infrastructure as code, in this scenario that happens to be using AWS CDK.
```

Comment:
This section sounds weird.. you talk in the first person then refer to "this scenario" rather than "in our case we were using AWS CDK" ... and perhaps link to what CDK is...